apps/supernova/src/lib/createFiltersSlice.jsx

```diff
@@ -28,11 +28,21 @@ const parsePredefinedFilters = (predefinedFilters) => {
 const parseInitialFilters = (initialFilters, filterLabels) => {
   if (!initialFilters) return initialFiltersState.initialFilters
 
-  if (typeof initialFilters !== "object") {
+  if (typeof initialFilters !== "object" || initialFilters === null) {
     console.warn("[supernova]::parseInitialFilters: initialFilters object is not an object")
     return {}
   }
 
+  // Check if all values are arrays
+  initialFilters = Object.entries(initialFilters).reduce((acc, [key, value]) => {
+    if (Array.isArray(value)) {
+      acc[key] = value // valid key-value pair
+    } else {
+      console.warn(`[supernova]::parseInitialFilters: Value for "${key}" is not an Array.`)
+    }
+    return acc
+  }, {})
+
   // Check if all keys are in filterLabelValues
   if (!Object.keys(initialFilters).every((key) => filterLabels.includes(key))) {
```

Comment:
**(Non-blocking concerns)**

Concerned that there's other string/array methods in this file that don't check valid type before executing e.g. `.included`.
Your changes look good, but if we've the time (even in a follow-up) PR would be great to ensure that we prevent the same issue causing bugs, at least in the same files.

Reply:
I see none. At least on the surface everything is tested. Maybe there are like here this problems in nested forms.